packages/product-json-to-csv/src/cli.js

```diff
@@ -0,0 +1,239 @@
+  })
+  .option('productSeparator', {
+    alias: 'ps',
+    default: '\n',
+    describe: 'Marker separating product chunks',
+    type: 'string',
+  })
+  .option('encoding', {
+    alias: 'e',
+    default: 'utf8',
+    describe: 'Encoding used when saving data to output file',
+    type: 'string',
+  })
+  .option('logLevel', {
+    default: 'info',
+    describe: 'Logging level: error, warn, info or debug',
+    type: 'string',
+  })
+  .option('prettyLogs ', {
+    describe: 'Pretty print logs to the terminal',
+    type: 'boolean',
+  })
+  .option('logFile', {
+    default: 'product-json-to-csv.log',
+    describe: 'Path to file where logs should be saved',
+    type: 'string',
+  }).argv
+
+// instantiate logger
+const loggerConfig = {
+  level: args.logLevel,
+  prettyPrint: args.prettyLogs,
+}
+const logger = pino(loggerConfig)
+
+// print errors to stderr if we use stdout for data output
+// if we save data to output file errors are already logged by pino
+const logError = error => {
+  const errorFormatter = new PrettyError()
+
+  if (logger.level === 'debug')
+    process.stderr.write(`ERR: ${errorFormatter.render(error)}`)
+  else process.stderr.write(`ERR: ${error.message || error}`)
+}
+
+const errorHandler = errors => {
+  if (Array.isArray(errors)) errors.forEach(logError)
+  else logError(errors)
+
+  process.exitCode = 1
+}
+
+// Retrieve the headers from the template file
+// Only the first line of the file is read
+const getHeaders = _args =>
+  new Promise((resolve, reject) => {
+    if (!_args.template) resolve(null)
+    else {
+      const rl = readline.createInterface({
+        input: _args.template,
+      })
+      rl.on('error', reject)
+      rl.on('line', line => {
+        rl.close()
+        resolve(line.split(_args.delimiter))
+      })
+    }
+  })
+
+const resolveCredentials = _args => {
+  if (_args.accessToken) return Promise.resolve({})
+  return getCredentials(_args.projectKey)
+}
+
+// If the stdout is used for a data output, save all logs to a log file.
+// pino writes logs to stdout by default
+if (args.output === process.stdout)
+  logger.stream = fs.createWriteStream(args.logFile)
+
+// Register error listener
+args.output.on('error', errorHandler)
+
+let headerFields
+getHeaders(args)
+  .then(headerFieldsfromTemplate => {
+    headerFields = headerFieldsfromTemplate
+    return resolveCredentials(args)
+  })
+  .then(credentials => {
+    const apiConfig = {
+      host: args.authUrl,
+      apiUrl: args.apiUrl,
+      projectKey: args.projectKey,
+      credentials,
+    }
+    const productJsonToCsvConfigOptions = {
+      categoryBy: args.categoryBy,
+      categoryOrderHintBy: args.categoryOrderHintBy,
+      delimiter: args.delimiter,
+      fillAllRows: args.fillAllRows,
+      language: args.language,
+      multiValueDelimiter: args.multiValueDelimiter,
+      productSeparator: args.productSeparator,
+      headerFields,
+    }
+    const myLogger = {
```

Comment:
Please rename. `myLogger` feels odd. `parserLogger` of whatever it belongs to might be a better option.